### ISSUE TITLE
Aer 0.5 not handling complex floats in state initialization

### ISSUE THREAD

**Issue by charleshadfield** [Qiskit/qiskit-aer#692]
### Informations

- **Qiskit AER version**: 0.5.0
- **Operating system**: mac 10.14.6

### What is the current behavior?

If a QuantumCircuit is executed with an initialize command, a RuntimeError is thrown whenever the state chosen to initialize has complex amplitudes. (This problem has occurred upon updating to `qiskit==0.18` it does not occur for `qiskit==0.17`)

### Steps to reproduce the problem

    from qiskit import QuantumCircuit, execute
    from qiskit import Aer
    # from qiskit.quantum_info import random_statevector
    import numpy as np
    
    simulator = Aer.get_backend('qasm_simulator')
    
    circ = QuantumCircuit(1,1)
    # state = random_statevector(2).data
    state = (1/np.sqrt(2))*np.array([1, 1+0j]) # the presence of 0j throws the error
    circ.initialize(state, 0)
    circ.measure(0,0)
    
    job = execute(circ, simulator, shots=1)
    result = job.result()

The error:

    RuntimeError: Unable to cast Python instance to C++ type (compile in debug mode for details)

### Suggested solutions

**Comment by chriseclectic** [comment]
The issue is that because of the way the terra `Instruction` checks param values the complex values of the array are being added to the params list as `numpy.complex128` instead of standard python `complex`, and these weren't caught by the pybind11 wrapper to C++. 

@ereastman is working on adding support for that to the wrappers, but probably the Instruction object should be modified to call `tolist` on Numpy arrays if it is storing the array as a param list, or better yet the param should store the full numpy vector since it is more efficient to pass Numpy arrays to C++ through Pybind11.

A work around in the meantime is to call `tolist` yourself like this:

```python
psi = np.array([1, -1j]) / np.sqrt(2)
circ.initiliaze(psi.tolist(), [0])
```

**Comment by chriseclectic** [comment]
This should be fixed by #697
